pytorch_lightning/utilities/cloud_io.py

```diff
@@ -14,15 +14,18 @@
 
 import io
 from pathlib import Path
-from typing import IO, Union
+from typing import Any, Dict, IO, Optional, Union
 
 import fsspec
 import torch
-from fsspec.implementations.local import LocalFileSystem
+from fsspec.implementations.local import AbstractFileSystem, LocalFileSystem
 from packaging.version import Version
 
 
-def load(path_or_url: Union[str, IO, Path], map_location=None):
+def load(
+    path_or_url: Union[str, IO, Path],
+    map_location: Optional[Union[str, torch.device, Dict[Union[str, torch.device], Union[str, torch.device]]]] = None,
```

Comment:
this can be a callable too in case of torch.load

Reply:
Done !